ReferenceError: _objectSpread is not defined after update

## Bug Report

**Current Behavior**
I've updated to 7.5 and have been following the developments with issues that should have been resolved with https://github.com/babel/babel/releases/tag/v7.5.1 however after stripping and reinstalling everything babel-related, I'm still getting this error. 

**Babel packages**
@babel/cli: "^7.5.0"
@babel/core: "^7.5.0"
@babel/helpers: "^7.5.1"
@babel/plugin-proposal-object-rest-spread: "^7.5.1"
@babel/plugin-transform-runtime: "^7.5.0"
@babel/polyfill: "^7.4.4"
@babel/preset-env: "^7.5.0"
@babel/preset-react: "^7.0.0"
babel-eslint: "^10.0.2"
babel-jest: "^24.8.0"
babel-loader: "^8.0.6"
babel-plugin-dynamic-import-node: "^2.3.0"
babel-plugin-jsx-remove-data-test-id: "^2.0.0"
babel-plugin-transform-class-properties: "^6.24.1"
babel-plugin-transform-decorators-legacy: "^1.3.5"
babel-plugin-transform-export-extensions: "^6.22.0"
babel-plugin-transform-object-rest-spread: "^6.26.0"

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "env": {
    "test": {
      "plugins": [
        "transform-class-properties",
        "dynamic-import-node",
        "@babel/plugin-transform-runtime"
      ]
    },
    "production": {
      "plugins": ["babel-plugin-jsx-remove-data-test-id"]
    }
  },
  "presets": ["@babel/preset-env", "@babel/preset-react"],
  "plugins": ["transform-class-properties"]
}

```

**Environment**
- Babel version(s): 7.5.0
- Node/npm version: 6.5.0
- OS: Linux Mint Cinnamon
- Monorepo: yes

If you delete those entries from yarn.lock and re-run `yarn --frozen-lockfile`, tests pass :thinking: 	

@v-angolenko I get this error:
```
➜ npm test

> @deepvision/svg-sprite-generator@0.1.3 test /home/nicolo/Documenti/dev/babel-bugs/spread-issue
> jest

sh: 1: jest: not found
npm ERR! Test failed.  See above for more details.
```

--- a/packages/babel-plugin-proposal-object-rest-spread/src/index.js
+++ b/packages/babel-plugin-proposal-object-rest-spread/src/index.js
@@ -455,6 +455,11 @@ export default declare((api, opts) => {
           try {
             helper = file.addHelper("objectSpread2");
           } catch {
+            // TODO: This is needed to workaround https://github.com/babel/babel/issues/10187
+            // and https://github.com/babel/babel/issues/10179 for older @babel/core versions
+            // where #10187 isn't fixed.
+            this.file.declarations["objectSpread2"] = null;
+
             // objectSpread2 has been introduced in v7.5.0
             // We have to maintain backward compatibility.
             helper = file.addHelper("objectSpread");

